caddytls/challenge_provider.go

```diff
@@ -0,0 +1,6 @@
+package caddytls
+
+import "github.com/xenolf/lego/acme"
+
+// ChallengeProvider type to be used in Caddy plugins over acme.ChallengeProvider directly, to avoid https://github.com/mattfarina/golang-broken-vendor
```

Comment:
Split this comment into multiple, shorter lines. Also, describe what it is/does, or link to the origin type's docs: https://godoc.org/github.com/xenolf/lego/acme#ChallengeProvider